pmlc/rt/vulkan/wrappers.cc

```diff
@@ -21,111 +21,207 @@
 #include "pmlc/rt/vulkan/vulkan_runtime.h"
 
 namespace {
-
 class VulkanRuntimeManager {
 public:
-  VulkanRuntimeManager() = default;
+  VulkanRuntimeManager() {
+    if (failed(vulkanRuntime.init())) {
+      llvm::errs() << "vulkanRuntime.init() failed";
+    }
+  }
   VulkanRuntimeManager(const VulkanRuntimeManager &) = delete;
   VulkanRuntimeManager operator=(const VulkanRuntimeManager &) = delete;
-  ~VulkanRuntimeManager() = default;
-
-  void setResourceData(DescriptorSetIndex setIndex, BindingIndex bindIndex,
-                       const VulkanHostMemoryBuffer &memBuffer) {
-    std::lock_guard<std::mutex> lock(mutex);
-    vulkanRuntime.setResourceData(setIndex, bindIndex, memBuffer);
+  ~VulkanRuntimeManager() {
+    if (failed(vulkanRuntime.destroy())) {
+      llvm::errs() << "vulkanRuntime.destroy() failed";
+    }
   }
 
-  void setEntryPoint(const char *entryPoint) {
+  void createLaunchKernelAction(uint8_t *shader, uint32_t size,
+                                const char *entryPoint,
+                                NumWorkGroups numWorkGroups) {
     std::lock_guard<std::mutex> lock(mutex);
+    vulkanRuntime.createLaunchKernelAction();
+    vulkanRuntime.setShaderModule(shader, size);
     vulkanRuntime.setEntryPoint(entryPoint);
+    vulkanRuntime.setNumWorkGroups(numWorkGroups);
```

Comment:
Why not just have the `createLaunchKernelAction` take the `shader`, `entryPoint` and `numWorkGroups`?

Reply:
Yes, I will add them to a single function. It is succinct and better. I thought to leave these functions as they are quite self-explanatory.